sdk/lib/opentelemetry/sdk/trace/span_limits.rb

```diff
@@ -33,23 +36,26 @@ class SpanLimits
         # @raise [ArgumentError] if any of the max numbers are not positive.
         def initialize(attribute_count_limit: Integer(ENV.fetch('OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT', 128)), # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
                        attribute_length_limit: ENV.fetch('OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT',
-                                                         ENV['OTEL_RUBY_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT']),
+                                                         ENV['OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT'] || ENV['OTEL_RUBY_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT']),
```

Comment:
I _believe_ we want to switch the order of the fallback values in `fetch` here. So if it's currently `ENV['OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT'] || ENV['OTEL_RUBY_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT']),`, we want it to be `ENV['OTEL_RUBY_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT']) || ENV['OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT']`. 

See:

```ruby
irb(main):001:0> nil || "foo" || "bar"
=> "foo"
```

This is based on the assumption that we want the `OTEL_RUBY_*` env vars to take precedence over `OTEL_*` env vars. Maybe we can add an explicit test case for this?

Side note: the Go SDK has a function dedicated to reconciling env vars in priority order that we _might_ want to emulate? https://github.com/open-telemetry/opentelemetry-go/pull/2677/files#diff-603faa80af2db68f66a19ef9a299eb128cd2e2d99c53506852d5107eb2a92a24R77-R97. I find the `ENV.fetch("foo", ENV["bar"] || ENV["baz"])` convention a bit hard to process mentally.

Reply:
In this case, the specced env var should trump the Ruby-specific one IMO.

Reply:
Either way is fine by me, so long as we document 😄